src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -662,201 +415,444 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
-                                    }}
-                                />
-                            </div>
-                        </Stack>
-
-                        <MiniMap
-                            nodeColor={theme.semanticColors.inputBackground}
-                            style={graphViewerMinimapStyles}
-                        />
-                        <Controls
-                            className={graphViewerStyles.graphViewerControls}
-                        />
-                        <Background
-                            color={theme.semanticColors.bodyBackground}
-                            gap={16}
-                            onClick={onBackgroundClick}
-                        />
-                    </ReactFlow>
-                </ElementsContext.Provider>
+                                <OATModelList />
+                            </Callout>
+                        )}
+
+                    <GraphViewerControls
+                        legendButtonId={legendButtonId}
+                        miniMapButtonId={mapButtonId}
+                        modelListButtonId={modelListButtonId}
+                        onApplyAutoLayoutClick={forceGraphLayout}
+                    />
+
+                    {oatGraphState.isMiniMapVisible && (
+                        <div className={classNames.graphMiniMapContainer}>
+                            <MiniMap
+                                nodeColor={'transparent'}
+                                nodeStrokeColor={theme.palette.black}
+                                nodeStrokeWidth={2}
+                                maskColor={getControlBackgroundColor(theme)}
+                                className={classNames.graphMiniMap}
+                            />
+                        </div>
+                    )}
+                    {oatGraphState.isLegendVisible && !oatGraphState.isLoading && (
+                        <Callout
+                            setInitialFocus={true}
+                            styles={classNames.subComponentStyles.legendCallout}
+                            target={`#${legendButtonId}`}
+                            directionalHint={DirectionalHint.topRightEdge}
+                        >
+                            <GraphLegend />
+                        </Callout>
+                    )}
+                    <Background gap={16} onClick={clearSelectedModel} />
+                </ReactFlow>
             </div>
         </ReactFlowProvider>
     );
 };
 
-OATGraphViewer.defaultProps = {
-    importModels: []
+const OATGraphViewer: React.FC<IOATGraphViewerProps> = (props) => {
+    return (
+        <OatGraphContextProvider>
+            <OATGraphViewerContent {...props} />
+        </OatGraphContextProvider>
+    );
 };
 
-export default OATGraphViewer;
+export default styled<
+    IOATGraphViewerProps,
+    IOATGraphViewerStyleProps,
+    IOATGraphViewerStyles
+>(OATGraphViewer, getStyles);
```

Comment:
It would be good to see if this could be refactored into smaller files to ease future reviews.

Reply:
The graphViewer is unfortunately kinda complex. I tried to pull out anything not directly related to the rendering of the graph library or building the data it needs to render. If you see particular parts that look like potentials, I'm open to looking into that. Believe it or not, this is already remarkably more simple and concise than it used to be. :D but I'm still willing to give it a go. I guess we could try to pull some of the handlers or data transformations into utils but right now it's at least possible to trace through things, I worry that all we would gain from refactoring might be to make it more convoluted. 🤷